api/v1alpha1/oauth2client_types.go

```diff
@@ -265,6 +270,13 @@ const (
 	OAuth2ClientConditionReady = "Ready"
 )
 
+// OAuth2ClientDeletionPolicy represents if a deleted oauth2 client object should delete the database row or not.
+type OAuth2ClientDeletionPolicy string
+
+const (
```

Comment:
In general I have no issues with the changes and tests :) The only thing I would modify is to change the type to use iota and specify the valid values.
Also, please run the generation makefile commands to generate the api schemas :)

Reply:
Hey @Demonsthere 👋 

Sure, I pushed some changes.
I hope everything is up to date now :)